pkg/interactor/list_data.go

```diff
@@ -10,94 +10,55 @@ import (
 
 // List provides all the thing's data owned by an user
 func (d *DataInteractor) List(token string, query *entities.Query) ([]entities.Data, error) {
-	data, err := d.getDevicesData(token, query)
+	things, err := d.things.List(token)
 	if err != nil {
-		return []entities.Data{}, err
+		return nil, fmt.Errorf("error getting list of things: %w", err)
 	}
 
-	data, err = d.filterDataBySensorID(data, query.ThingID, query.SensorID)
-	if err != nil {
-		return []entities.Data{}, err
+	if query.ThingID != "" {
+		things = []*btEntities.Thing{{ID: query.ThingID}}
```

Comment:
Did you test all the endpoints? Notice that we should return all the data from the things whose owner is a single user. That's what `getDevicesData` tried to do.

Reply:
I have tested with storage SDK with all three methods: `listData`, `listDataByDevice` and `listDataBySensor`. But I gonna make more intensive tests

Reply:
@esdrasjnr I think you can put the results on the PR description so that I can easily analyse.